src/__tests__/calculate.test.js

```diff
@@ -0,0 +1,17 @@
+import calculate from '../logic/calculate';
+
+test('should reset the calculator when AC button is pressed', () => {
+  const initialState = {
+    total: '123',
+    next: '456',
+    operation: '+',
+  };
+
+  const result = calculate(initialState, 'AC');
+
+  expect(result).toEqual({
+    total: null,
+    next: null,
+    operation: null,
+  });
+});
```

Comment:
- [ ] Well done for covering some of the functionalities and scenarios of the `calculate.js` module 👏. However, all the functionalities and scenarios need to be covered using Jest as per requirements. Please consider writing more tests 👍